volatility/framework/plugins/mac/pslist.py

```diff
@@ -258,41 +244,11 @@ def list_tasks_process_group(cls,
                                    subtype = kernel.get_type("pgrphashhead"))
 
         for proc_list in proc_array:
-            # test the validity of the current element
-            # it is expected that many won't be initialized
-            try:
-                pgrp = proc_list.lh_first
-            except exceptions.InvalidAddressException:
-                continue
-
-            seen_pgrps = set()
-
-            # this walks the particular process group
-            while pgrp and pgrp.vol.offset not in seen_pgrps:
-                seen_pgrps.add(pgrp.vol.offset)
-
-                # nothing can be done if this list pointer is invalid, so move on
-                try:
-                    p = pgrp.pg_members.lh_first
-                except exceptions.InvalidAddressException:
-                    break
-
-                seen_pg = set()
-                while p and p.vol.offset not in seen_pg:
-                    seen_pg.add(p.vol.offset)
-
-                    if p.is_readable() and not filter_func(p):
+            for pgrp in mac.MacUtilities.walk_list_head(proc_list, "pg_hash"):
+                for p in mac.MacUtilities.walk_list_head(pgrp.pg_members, "p_pglist"):
```

Comment:
Not entirely happy about just p as a variable name, but it's used immediately after, so I'm not gonna quibble about it too much, just point it out and mebbe look over the top of my glasses at you...  5;P

Reply:
@ikelos <off-topic/> I really like your style. This is exactly what distinguishes good projects from awesome projects.

Reply:
good call. Changed these